pkg/graveler/staging/db_manager.go

```diff
@@ -113,3 +137,81 @@ func (p *DBManager) txOpts(opts ...db.TxOpt) []db.TxOpt {
 	}
 	return append(o, opts...)
 }
+
+// runWorker writing staging token data to fd
+func runWorker(ctx context.Context, d *pgxpool.Pool, jobChan <-chan string) error {
+	record := &graveler.ValueRecord{}
+	for t := range jobChan {
+		stagingToken := t
+		rows, err := d.Query(ctx, "SELECT key, identity, data FROM graveler_staging_kv where staging_token=$1", stagingToken)
+		if err != nil {
+			return err
+		}
+		scanner := pgxscan.NewRowScanner(rows)
+		for rows.Next() {
+			err = scanner.Scan(record)
+			if err != nil {
+				return err
+			}
+			pb := protoFromValue(record.Key, record.Value)
+			data, err := proto.Marshal(pb)
+			if err != nil {
+				return err
+			}
+			if err = encoder.Encode(kv.Entry{
+				PartitionKey: []byte(stagingToken),
+				Key:          record.Key,
+				Value:        data,
+			}); err != nil {
+				return err
+			}
+		}
+		rows.Close()
```

Comment:
`defer rows.close()` right after the query and err check

Reply:
@itai-david  Thanks for the review!! Please note that the rows variable is created inside a for loop, so the common practice is to use close and not defer.
We can export this logic to function, but I think it's safe to say that the migration process will not cause a resource leak in the server (it is executed separately and we don't handle panics)